pygdf/dataframe.py

```diff
@@ -1038,6 +1038,36 @@ def hash_columns(self, columns=None):
         cols = [self[k]._column for k in columns]
         return Series(numerical.column_hash_values(*cols))
 
+    def partition_by_hash(self, columns, nparts):
+        """Partition the dataframe by the hashed value of data in *columns*.
+
+        Parameters
+        ----------
+        columns : sequence of str
+            The names of the columns to be hashed.
+            Must have at least one name.
+        nparts : int
+            Number of output partitions
+
+        Returns
+        -------
+        partitioned: list of DataFrame
+        """
+        cols = [col._column for col in self._cols.values()]
+        names = list(self._cols.keys())
+        key_indices = [names.index(k) for k in columns]
```

Comment:
Couldn't this calculation of `key_indices` happen inside the function, or is there a reason to leave to the caller/user?

Reply:
Is this function going to be used for anything besides the dask multicolumn join in the near future? If not then your judgement call is as good as any tbh

Reply:
there're no plan to use it in other things.   dask multicol join is the primary driver at this point.